pulpcore/pulpcore/app/models/repository.py

```diff
@@ -212,6 +212,8 @@ class RepositoryVersion(Model):
     repository = models.ForeignKey(Repository, on_delete=models.CASCADE)
     number = models.PositiveIntegerField(db_index=True)
     complete = models.BooleanField(db_index=True, default=False)
+    base_version = models.ForeignKey('Repositoryversion', null=True,
+                                     on_delete=models.SET_NULL)
```

Comment:
so we won't propagate delete of base_version but we store diff of added vs removed artifacts, is this cool?

Reply:
makes sense. it allows i.e to set the base version couple of times, resetting the initial content set again.